VerilogCodingStyle.md

```diff
@@ -1745,6 +1756,18 @@ an always block as occurring in a separate simulation event as the non-blocking
 assignment. This process makes some signals jump registers, potentially leading
 to total protonic reversal. That's bad.
 
+Exception: For a clock divider blocking assignment must be used not to cause race condition.
```

Comment:
This one is new to me, can you provide a citation?

Reply:
I don't really understand the problem, sorry. Both `q_nb_ng` and `q_b_ok` are incrementing in steps of 2, as I would expect.

You're right that they sample differently: `q_nb_ng` gets the value after clk and `q_b_ok` gets the one before. But that's not a race condition: I think those semantics are guaranteed. Do you think there's a valid schedule where `q_nb_ng` gets the value `0010` at the end of this clock tick?

Incidentally, you're right about the "time slot A / time slot B" thing, sorry. I was thinking about the original Verilog scheduling semantics, which is defined in terms of triggering processes. There, you'd get the processes triggered one after the other (with no time between them, giving delta cycles). The SystemVerilog scheduling gives the same result by defining a single time slot that jumps back from NBA to active.

Reply:
@rswarbrick,  

It seems that I could explain the difficult part of this issue to you.

> I think those semantics are guaranteed.

You are right.  I used the word "race condition" for one between clk and clk_div_nb.

q_nb_ng and q_b_ok are D-Flip-flops which capture d (the output of D-Flip-flops in  clock domain clk).  They have to capture value 0b0010 as q_b_ok does in the case above.